index.js

```diff
@@ -9,38 +9,43 @@ const got = require('got');
 const get = require('lodash.get');
 
 class Seeder {
-  constructor(logHandler, showSpinner) {
-    this.log = logHandler;
-    this.showSpinner = showSpinner
-  }
-
-  async run(path = process.cwd()) {
-    const entries = Object.entries(this.parsePath(path.replace(new RegExp(`${sep}$`), '')));
-
-    if (entries.length === 0) {
-      return this.error('The given path must be a .json file or a directory containing at least one valid .json file');
-    }
-
-    entries.sort(([a], [b]) => a === 'assets' ? 1 : b === 'assets' ? -1 : 0);
-
-    if (this.showSpinner) {
-        this.spinner = ora({
-          text: 'Seeding. This may take some time...',
-          stream: process.stdout,
-        }).start();
-    }
-
-    const typeCounts = {};
-    const queuedPromises = [];
-    const responses = {};
-    const queue = new PromiseQueue(1, Infinity);
-
-    entries.forEach(([endpoint, data]) => {
-      (Array.isArray(data) ? data : [data]).forEach((datum, index) => {
-        const {link, ...rest} = datum;
-
-        queuedPromises.push(queue.add(() =>
-          this.post(`/v1/${endpoint}`, rest)
+	constructor(logHandler, showSpinner) {
+		this.log = logHandler;
+		this.showSpinner = showSpinner
+	}
+
+	async run(path = process.cwd()) {
+		const entries = Object.entries(this.parsePath(path.replace(new RegExp(`${sep}$`), '')));
+
+		if (entries.length === 0) {
+			return this.error('The given path must be a .json file or a directory containing at least one valid .json file');
+		}
+
+		entries.sort(([a], [b]) => a === 'assets' ? 1 : b === 'assets' ? -1 : 0);
+
+		if (this.showSpinner) {
+			this.spinner = ora({
+				text: 'Seeding. This may take some time...',
+				stream: process.stdout,
+			}).start();
+		}
+
+		const typeCounts = {};
+		const queuedPromises = [];
+		const responses = {};
+		const queue = new PromiseQueue(1, Infinity);
+
+		entries.forEach(([endpoint, data]) => {
+			(Array.isArray(data) ? data : [data]).forEach((datum, index) => {
+				const {link, ...rest} = datum;
+
+				queuedPromises.push(queue.add(() => {
+				  if (this.showSpinner) {
+				    this.spinner.stop();
+				    this.spinner.start(`Seeding ${chalk.dim(endpoint)} #${index}. This may take some time...`);
```

Comment:
Pretty sure here you can just `this.spinner.text = ...`. Also, the `PromiseQueue` usage means that we can probably add some concurrency to this in the future. Not sure if we want to consider that now.

Reply:
I've updated to use `.text = '...`. I've put the spinner change into the promise that calls the API, so hopefully concurrency would still be alright here. It might get a bit whacky with what you actually see, but we can deal with that if we add concurrent handlers for the promise queue perhaps?